File: DAL/Dtos/ApartmentDto.cs
namespace DAL.Dtos
{
    public class ApartmentDto
    {
        public int Id { get; set; }

        public string? RelatedFamily { get; set; }

        public int RegisteredCountPeople { get; set; }

        public string? ReceiptCode { get; set; }

        public AddressDto? Address { get; set; }

        public MeterDocumentDto? MeterDocument { get; set; }

        public IEnumerable<MeterDto>? Meters { get; set; }
    }
}

File: Utilities/ClientApp/src/app/core/Models/ApartmentDto.ts
import { AddressDto } from "./AddressDto";
import { MeterDto } from "./MeterDto";

export interface ApartmentDto {
    id?: number;
    relatedFamily: string;
    registeredCountPeople: number;
    receiptCode: string;
    address: AddressDto;
    meters?: MeterDto[];
}